Users should be warned of Apache/GPLv2 compatibility concerns

ncoghlan opened the pull request:
Choosing to use the Apache license instead of MIT means that some "GPLv2 only" projects may choose not to use the software. This is rarely something that users of a permissive license actually want.

While the legal situation is murky (the ASF and FSF disagree about whether the licenses are actually incompatible), this is not just a theoretical problem from a social perspective: https://github.com/twitter/bootstrap/issues/2054

ncoghlan committed:
Users should only choose APL if they're not worried about GPL compatibility

Choosing to use APL instead of MIT means that "GPLv2 Only" projects cannot use the software. This is rarely something that users of a permissive license actually want.

This is not just a theoretical problem: https://github.com/twitter/bootstrap/issues/2054

gstein commented:
There is no "APL" ... you likely mean Apache License, v2.0. Or "ALv2".

And note that the "theoretical" incompatibility is a statement/belief of the FSF. The Apache Software Foundation does **not** believe they are incompatible.

ncoghlan committed:
Note possible social consequences of choosing Apache license

For people choosing a license, an important thing to consider when choosing the Apache license over MIT is that they need to see if there are any GPLv2 projects they want to use their software that takes a "better safe than sorry" approach to the disagreement between the FSF and ASF about the compatibility of GPLv2 and the Apache license.

It seems unlikely that there could ever be a serious legal dispute about this (since any incompatibility clearly hangs on a legal technicality rather than on the *intent* of the two licenses), but the *social* consequences are independent of the legal ones.

ncoghlan commented:
Thanks for pointing that out, Greg (both the incorrect acronym and the fact that the ASF disagrees with the FSF's assessment that the licenses are incompatible).

I've updated the description and the proposed change to the home page accordingly, attempting to highlight that this is more of a social issue, rather than a legal one.

dalb8 commented:
This is particularly important in the case of Android apps, 95% of which contain some Apache licensed code as the support libraries at least are under that licence.

If the copyright header suggestions become more explicit the option of GPLv2+ can be mentioned, allowing for the entire work to be distributed as GPLv3.

haacked commented:
> I've updated the description and the proposed change to the home page accordingly, attempting to highlight that this is more of a social issue, rather than a legal one.

I'm hesitant to include this on the front page. There are probably many "social issues" we could get into. For example, I can name very large influential tech companies (ok company) that refuse to incorporate GPL because they feel it is "viral". Should we also include a note on that?

Perhaps we need to think about a way to add notes to the actual license page or another more "holistic" approach to bringing out this type of peripheral information.
